src/Jackalope/Node.php

```diff
@@ -1656,8 +1683,9 @@ public function beginTransaction()
         parent::beginTransaction();
 
         // Notify the children properties
-        foreach ($this->properties as $prop) {
-            $prop->beginTransaction();
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
I don't really know... Shouldn't a property that is added later have the same transaction state? However, if this is removed, it should also be removed in the `commitTransaction` and `rollbackTransaction` methods, right?